superset/viz.py

```diff
@@ -474,17 +480,30 @@ def get_payload(self, query_obj: Optional[QueryObjectDict] = None) -> VizPayload
         filters = self.form_data.get("filters", [])
         filter_columns = [flt.get("col") for flt in filters]
         columns = set(self.datasource.column_names)
+        filter_values_columns = []
+
+        # if using virtual datasource, check filter_values
+        if self.datasource.sql:
+            statements_without_comments = sqlparse.format(
```

Comment:
Because we're doing regex of the SQL statement (and not actual parsing of the statement) this seems somewhat of a hybrid approach. I would likely either:

1. Don't worry about parsing the statement and just run the regex logic on the full SQL statement.
2. Leverage `sqlparse` to actually identify the `filter_value` function and the items in the identifier list, i.e., 

```
stmt._pprint_tree()
...
      |     |- 7 Function 'filter...'
      |     |  |- 0 Identifier 'filter...'
      |     |  |  `- 0 Name 'filter...'
      |     |  `- 1 Parenthesis '('my_d...'
      |     |     |- 0 Punctuation '('
      |     |     |- 1 IdentifierList ''my_da...'
      |     |     |  |- 0 Single ''my_da...'
      |     |     |  |- 1 Punctuation ','
      |     |     |  |- 2 Whitespace ' '
      |     |     |  `- 3 Single ''birth...'
      |     |     `- 2 Punctuation ')'
...
```

Reply:
thanks John! I didn't know the sql parser is this powerful. But i will go with solution 1). I just feel traverse the statement tree for this small thing is overkill (and probably slower than simple regex).

Reply:
I agree.